app/connection.go

```diff
@@ -48,6 +48,7 @@ func GetServerConnection(c *cli.Context, opts ...grpc.DialOption) (context.Conte
 
 	ctx = metadata.AppendToOutgoingContext(ctx, "tcld-version", getVersion())
 	ctx = metadata.AppendToOutgoingContext(ctx, "tcld-commit", Commit)
+	ctx = metadata.AppendToOutgoingContext(ctx, "temporal-cloud-api-version", "2023-07-26-01")
```

Comment:
I haven't successfully tested the proto fileoption approach, so use this to start testing first, and will have follow up PRs to get the proto fileoption integrated.